src/GenericFactory.sol

```diff
@@ -7,13 +7,28 @@ import { SSTORE2 } from "solady/utils/SSTORE2.sol";
 import { Owned } from "solmate/auth/Owned.sol";
 
 import { IGenericFactory } from "src/interfaces/IGenericFactory.sol";
+import { StableMintBurn } from "src/curve/stable/StableMintBurn.sol";
 
 uint256 constant MAX_SSTORE_SIZE = 0x6000 - 1;
 
 contract GenericFactory is IGenericFactory, Owned {
-    bool private _deployInProgress = false;
+    constructor(address aOwner) Owned(aOwner) {
+        bytes memory lInitCode = type(StableMintBurn).creationCode;
 
-    constructor(address aOwner) Owned(aOwner) { } // solhint-disable-line no-empty-blocks
+        address lStableMintBurn;
+        // SAFETY:
+        // Does not write to memory
+        assembly ("memory-safe") {
+            // sanity checked against OZ implementation:
+            // https://github.com/OpenZeppelin/openzeppelin-contracts/blob/3ac4add548178708f5401c26280b952beb244c1e/contracts/utils/Create2.sol#L40
+            lStableMintBurn := create2(callvalue(), add(lInitCode, 0x20), mload(lInitCode), 0)
+
+            if iszero(extcodesize(lStableMintBurn)) { revert(0, 0) }
+        }
+        emit Deployed(lStableMintBurn);
+    }
```

Comment:
actually, we can just use the solidity syntax for this instead of the assembly one right ,since we don't have to do the memory manipulation anymore?

Reply:
Correct. We can just do `new StableMintBurn{salt: bytes32(0)}();`

Reply:
okay done